### ISSUE TITLE
SUF-1-300

### ISSUE THREAD

**Issue by ivan-mezentsev** [SergejPr/NooLite-F#1]
How about support the subject?

**Comment by sprytkov** [comment]
Hi,

I will check it in this weekend's. I think that it should be worked without any issues.

**Comment by ivan-mezentsev** [comment]
I asked about "read_state" for brightness NOOLITE_F...

**Comment by SergejPr** [comment]
Unfortunatelly, I can't do the full test of this module. I can't switch it to dimmer mode. In manual is specify that this modul can be switched to dimmer mode via command from adapter, but this command is missed in API documentation. I wait answer from NooLite support. Also NooLite One v 1.5 app also doesn't recognize this module, so I can't use this app to turn on dimmer mode.

**Comment by ivan-mezentsev** [comment]
I’ve got it! The answer it:
“Можно вручную на блоке: длительным удержанием сервисной кнопки. держим кнопку, блок входит в режим очистки памяти, кнопку не отпускаем, держим дальше, диод гаснет потом, через секунд 30 светодиод начнет светиться постоянно-отпускаем кнопку и один раз кратковременно ее нажимаем.”

**Comment by SergejPr** [comment]
Thanks, 

I already switched it programmatically. I have used SLF-1-300 module to grab commands for it configuration, and then tried to set unused bits (for SLF the dimmer mode settings are disabled in app, so I couldn't grap it). As result I found appropriate combination. 

The command for module configuration:

```
CMD:129 FM:16 D0:xx D1:0 D2:127 D3:0

bits for D0:
0: save the module state when power is down
1: dimmer mode
2: allow accept noolite commands
3-4: extra button configuration (00 - switch mode, 01 - button, 10 - breaker??, 11 - disable extra button)
5: state after power up (has effect only if the saving of module state is disabled) 
6: retranslation of the  nooLite commands
7: ???? 
```

**Note:** the byte D3 should has value 127, otherwise configuration is not save. May be it is a bit mask to specify bits from D0 for saving, I haven't checked this. 


So I did quick test and noticed some issues with set_brightness and read_state command.

1. if I try to turn on module via set_brightness command with value 1.0 the module won't turn on. This works for SU-1-500, but it is broken for SUF-1-300.

Seems this issue is related with values that I send to the module.

By specs:
> При передаче команды со значением=6 (set_brightness)  и «Формат» =1 в данном байте содержится информация о яркости, которая будет установлена (значение в диапазоне 35…155). При значении 0 – свет выключится, при значении больше 155 – свет включится на максимальную яркость. 


By the fact:
SU-1-500 works as describd above:
- for 0 it turns off.  
- for values from 1 to 34 the lamp has minimal brightness (but it is n't specify)
- for 155..255 it has max brightness

but SUF-1-300 has different behavior:
- for 0..27 it turns off,
- for values in ranges 28..34 and 156..255 it doesn't have any reaction

And this behavior causes this issue, because I convert 1.0 value into 255 (not into 155).

2. the brightness value that returned in read_state command is some less than I set, for example I set brightness to 0.99, but in read_state I get 0.98. I think that these issue related with range conversion that I do in the code. For to set brightness we use 35..155 range (see specs above), but brightness in read_state response is in 0..255 range.

I think I will fix these issues in this weekends.  

P.S.  Можно и по-русски писать если что ;)
